x/twap/types/keys.go

```diff
@@ -79,8 +85,11 @@ func ParseTimeFromHistoricalTimeIndexKey(key []byte) time.Time {
 func ParseTimeFromHistoricalPoolIndexKey(key []byte) (time.Time, error) {
 	keyS := string(key)
 	s := strings.Split(keyS, KeySeparator)
-	if len(s) != 5 || s[0] != historicalTWAPPoolIndexNoSeparator {
-		return time.Time{}, fmt.Errorf("Called ParseTimeFromHistoricalPoolIndexKey on incorrectly formatted key: %v", s)
+	if len(s) != expectedLenOfKeySeparators {
+		return time.Time{}, fmt.Errorf("key incorrect length: expected length (%d), actual length (%d)", expectedLenOfKeySeparators, len(s))
+	}
+	if s[0] != historicalTWAPPoolIndexNoSeparator {
+		return time.Time{}, fmt.Errorf("historicalTWAPTimeIndexNoSeparator not formatted correctly: expected separator (%s), actual separator (%v)", historicalTWAPPoolIndexNoSeparator, s[0])
```

Comment:
Same as previous comment, this achieves the same as what it replaces but now passes mutation tests and gives off specific errors (which can then be specifically tested for)